streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -588,6 +592,29 @@ public void testCloseIsIdempotent() {
             closeCount, MockMetricsReporter.CLOSE_COUNT.get());
     }
 
+    @Test
+    public void testAddThread() {
+        props.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1);
+        final KafkaStreams streams = new KafkaStreams(getBuilderWithSource().build(), props, supplier, time);
+        streams.start();
+        final int oldSize = streams.threads.size();
+        try {
+            TestUtils.waitForCondition(() -> streams.state() == KafkaStreams.State.RUNNING, 15L, "wait until running");
+        } catch (final InterruptedException e) {
+            e.printStackTrace();
+        }
+        assertThat(streams.addStreamThread(), equalTo(Optional.of("newThread")));
+        assertThat(streams.threads.size(), equalTo(oldSize + 1));
+    }
+
+    @Test
+    public void testAddThreadNotDuringStart() {
```

Comment:
I would prefer to use `shouldNotAddThread()` as name although the pattern is different for the other test methods.